src/Microsoft.TestPlatform.CrossPlatEngine/EventHandlers/TestRequestHandler.cs

```diff
@@ -240,10 +240,11 @@ public void OnMessageReceived(object sender, MessageReceivedEventArgs messageRec
                     {
                         EqtTrace.Info("Discovery Session Initialize.");
                         this.testHostManagerFactoryReady.Wait();
+                        var discoveryEventsHandler = new TestDiscoveryEventHandler(this);
                         var pathToAdditionalExtensions = this.dataSerializer.DeserializePayload<IEnumerable<string>>(message);
                         jobQueue.QueueJob(
                                 () =>
-                                testHostManagerFactory.GetDiscoveryManager().Initialize(pathToAdditionalExtensions), 0);
+                                testHostManagerFactory.GetDiscoveryManager().Initialize(pathToAdditionalExtensions, discoveryEventsHandler), 0);
```

Comment:
>discoveryEventsHandler [](start = 116, length = 22)

should be testExtensionInitializeEventsHandler

Reply:
I will go ahead with this once we finalize the permanent fix for it